File: R/dados.R
#' Carregamento dos dados brutos do campeonato.
#'
#' Sem transformacao de negocio aqui -- so leitura e validacao estrutural do
#' CSV. As colunas de gols dos jogos ainda nao realizados ficam como `NA`
#' (`read.csv` ja faz essa conversao automaticamente ao ler campos vazios).

COLUNAS_ESPERADAS <- c(
  "rodada", "time_mandante", "gols_mandante", "time_visitante", "gols_visitante"
)

#' Le o CSV de jogos do campeonato.
#'
#' @param caminho caminho para o `brasileirao_2026.csv` (ex.: `data/raw/...`).
#' @return data.frame com uma linha por jogo. Jogos ainda nao realizados tem
#'   `gols_mandante`/`gols_visitante` como `NA`.
carregar_jogos <- function(caminho) {
  jogos <- read.csv(caminho, stringsAsFactors = FALSE)

  faltando <- setdiff(COLUNAS_ESPERADAS, names(jogos))
  if (length(faltando) > 0) {
    stop(sprintf("Colunas faltando no CSV: %s", paste(sort(faltando), collapse = ", ")))
  }

  jogos
}

#' Filtra apenas os jogos que ja tem resultado (gols preenchidos).
#'
#' @param jogos data.frame de jogos (ver `carregar_jogos()`).
#' @return subconjunto de `jogos` sem `NA` em `gols_mandante`/`gols_visitante`.
jogos_disputados <- function(jogos) {
  jogos[!is.na(jogos$gols_mandante) & !is.na(jogos$gols_visitante), ]
}

#' Filtra os jogos ainda nao realizados (gols em branco) -- o que precisa
#' ser simulado.
#'
#' @param jogos data.frame de jogos (ver `carregar_jogos()`).
#' @return subconjunto de `jogos` com `gols_mandante` igual a `NA`.
jogos_pendentes <- function(jogos) {
  jogos[is.na(jogos$gols_mandante), ]
}

File: src/brasileiro_mc/dados.py
"""Carregamento dos dados brutos do campeonato.

Sem transformação de negócio aqui — só leitura e validação estrutural do
CSV. As colunas de gols dos jogos ainda não realizados ficam como NaN
(pandas já faz essa conversão automaticamente ao ler o CSV).
"""

from __future__ import annotations

import pandas as pd

COLUNAS_ESPERADAS = [
    "rodada",
    "time_mandante",
    "gols_mandante",
    "time_visitante",
    "gols_visitante",
]


def carregar_jogos(caminho: str) -> pd.DataFrame:
    """Lê o CSV de jogos do campeonato.

    Args:
        caminho: caminho para o `brasileirao_2026.csv` (ex.: `data/raw/...`).

    Returns:
        DataFrame com uma linha por jogo. Jogos ainda não realizados têm
        `gols_mandante`/`gols_visitante` como `NaN`.

    Raises:
        ValueError: se alguma coluna esperada estiver faltando no CSV.
    """
    jogos = pd.read_csv(caminho)

    faltando = set(COLUNAS_ESPERADAS) - set(jogos.columns)
    if faltando:
        raise ValueError(f"Colunas faltando no CSV: {sorted(faltando)}")

    return jogos


def jogos_disputados(jogos: pd.DataFrame) -> pd.DataFrame:
    """Filtra apenas os jogos que já têm resultado (gols preenchidos)."""
    return jogos.dropna(subset=["gols_mandante", "gols_visitante"])


def jogos_pendentes(jogos: pd.DataFrame) -> pd.DataFrame:
    """Filtra os jogos ainda não realizados (gols em branco) — o que precisa
    ser simulado."""
    return jogos[jogos["gols_mandante"].isna()]
